Sources/SpeziScheduler/Scheduler.swift

```diff
@@ -109,31 +98,26 @@ public class Scheduler<Context: Codable>: NSObject, UNUserNotificationCenterDele
         if await !localNotificationAuthorization {
             try await UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound])
             
-            // Triggers an update of the UI in case the notification permissions are changed
-            await sendObjectWillChange()
+            // now we have permissions, schedule all notifications now
+            await updateScheduleNotifications()
         }
     }
     
     /// Schedule a new ``Task`` in the ``Scheduler`` module.
     /// - Parameter task: The new ``Task`` instance that should be scheduled.
+    @MainActor
     public func schedule(task: Task<Context>) async {
-        task.objectWillChange
-            .sink {
-                _Concurrency.Task {
-                    await self.sendObjectWillChange()
-                }
-            }
-            .store(in: &cancellables)
-        
-        tasks.append(task)
-        
-        self.updateTasks()
+        taskList.append(task)
+
+        for event in task.events { // make sure
```

Comment:
Not sure if this is a leftover of a thought:
```suggestion
        for event in task.events {
```